Sources/UIExtensions/ViewModifiers/View+HorizontalMask.swift

```diff
@@ -0,0 +1,39 @@
+// Copyright © 2022 Lautsprecher Teufel GmbH. All rights reserved.
+
+import SwiftUI
+
+extension View {
+
+    /// Adds a horizontally fade in/out effect to the view. The padding allows to start the fade effect relative to the horizontal bounds of the view.
+    public func maskHorizontally(leftPadding: CGFloat = 0, leftFade: CGFloat? = nil, rightFade: CGFloat? = nil, rightPadding: CGFloat = 0) -> some View {
+        self.mask(horizontalMask(leftPadding: leftPadding, leftFade: leftFade, rightFade: rightFade, rightPadding: rightPadding))
+    }
+
+    private func horizontalMask(leftPadding: CGFloat, leftFade: CGFloat? = nil, rightFade: CGFloat? = nil, rightPadding: CGFloat) -> some View {
+        HStack(spacing: 0) {
+            Rectangle()
+                    .frame(width: leftPadding)
+                    .opacity(0)
+
+            leftFade.map {
+                LinearGradient(gradient: Gradient(colors: [Color.black.opacity(0), Color.black]),
+                        startPoint: .leading,
+                        endPoint: .trailing)
+                        .frame(width: $0)
+            }
+
+            Color.black
```

Comment:
Can we get the colour injected? Color.black does never adapt to light/dark mode and this may be desired.

Reply:
Black is used to create a mask for the view (like a layer-mask in image editing)

Reply:
Ah, right, https://developer.apple.com/documentation/swiftui/form/mask(_:)
It makes sense.